src/pages/workspaces/workspace/Dashboard/Dashboard.ts

```diff
@@ -0,0 +1,662 @@
+import { displayConsentCodes, displayLibraryAttributes } from 'src/pages/workspaces/workspace/library-attributes';
+import { InitializedWorkspaceWrapper as Workspace, StorageDetails } from 'src/pages/workspaces/workspace/useWorkspace';
+import { wrapWorkspace } from 'src/pages/workspaces/workspace/WorkspaceContainer';
+
+const styles: Record<string, CSSProperties> = {
+  authDomain: {
+    padding: '0.5rem 0.25rem',
+    marginBottom: '0.25rem',
+    backgroundColor: colors.dark(0.15),
+    ...Style.noWrapEllipsis,
+  },
+  tag: {
+    padding: '0.25rem',
+    margin: '0.15rem',
+    backgroundColor: colors.dark(0.15),
+    borderRadius: 10,
+    overflow: 'hidden',
+    wordWrap: 'break-word',
+  },
+};
+
+const displayAttributeValue = (v: unknown): string => {
+  return cond(
+    [_.isArray(v), () => (v as string[]).join(', ')],
+    [v === true, () => 'Yes'],
+    [v === false, () => 'No'],
+    [typeof v === 'string', () => v as string],
+    [
+      !!v && typeof v === 'object' && 'items' in v && _.isArray(v.items),
+      () => ((v as any).items as string[]).join(', '),
+    ],
+    [typeof v === 'object', () => JSON.stringify(v)],
+    () => JSON.stringify(v)
+  );
+};
+
+const DataUseLimitations = ({ attributes }) => {
+  return _.map(
+    ({ key, title }) => {
+      return div({ key, style: { display: 'inline-block', marginRight: '0.75rem' } }, [
+        h(TooltipTrigger, { content: title }, [
+          span({ style: { textDecoration: 'underline dotted' } }, [key.slice(8)]),
+        ]),
+        ': ',
+        displayAttributeValue(attributes[key]),
+      ]);
+    },
+    _.filter(({ key }) => _.has(key, attributes), displayConsentCodes)
+  );
+};
+
+const DashboardAuthContainer = (props) => {
+  const { namespace, name } = props;
+  const { signInStatus } = useStore(authStore);
+  const [featuredWorkspaces, setFeaturedWorkspaces] = useState<{ name: string; namespace: string }[]>();
+
+  const isAuthInitialized = signInStatus !== 'uninitialized';
+
+  useEffect(() => {
+    const fetchData = async () => {
+      setFeaturedWorkspaces(await Ajax().FirecloudBucket.getFeaturedWorkspaces());
+    };
+    if (signInStatus === 'signedOut') {
+      fetchData();
+    }
+  }, [signInStatus]);
+
+  const isFeaturedWorkspace = () => _.some((ws) => ws.namespace === namespace && ws.name === name, featuredWorkspaces);
+
+  return cond(
+    [
+      !isAuthInitialized || (signInStatus === 'signedOut' && featuredWorkspaces === undefined),
+      () => centeredSpinner({ style: { position: 'fixed' } }),
+    ],
+    [signInStatus === 'signedOut' && isFeaturedWorkspace(), () => h(DashboardPublic, props)],
+    [signInStatus === 'signedOut', () => h(SignIn)],
+    () => h(WorkspaceDashboard, props)
+  );
+};
+
+interface RightBoxSectionProps {
+  title: string;
+  info?: ReactNode;
+  initialOpenState?: boolean;
+  afterTitle?: ReactNode;
+  onClick: () => void;
+}
+
+const RightBoxSection = (props: PropsWithChildren<RightBoxSectionProps>) => {
```

Comment:
Suggest including children in the props interface instead of adding it afterwards. This makes it possible to extend the interface if necessary.

```suggestion
const RightBoxSection = (props: RightBoxSectionProps) => {
```